draft-ietf-ipsecme-ikev2-multiple-ke-00.xml

```diff
@@ -520,21 +550,103 @@ Initiator                             Responder
+         |
+         +-- Transform AKE5 ( ID = NONE )
+                ]]></artwork></figure>
+
+                <t> In this example the initiator proposes to perform initial key exchange using 4096-bit MODP group
+                following by two mandatory additional key exchanges using FRODOKEM_976_AES and SIKE_P434 methods in any order,
+                following by additional key exchange using KYBER512 method that may be omitted.
+                </t>
+
+                <t> The responder might return the following SA payload, indicating that it agrees to 
+                perform two additional key exchanges SIKE_P434 followed by FRODOKEM_976_AES and doesn't
+                want to perform KYBER512 in addition.
+                </t>
+
+                <figure><artwork align="center" ><![CDATA[
+SA Payload
+   |
+   +--- Proposal #1 ( Proto ID = IKE(1), SPI size = 8,
+         |            6 transforms,      SPI = 0x8df52b331a196e7b )
+         |
+         +-- Transform ENCR ( ID = ENCR_AES_GCM_16 )
+         |     +-- Attribute ( Key Length = 256 )
+         |
+         +-- Transform KE ( ID = 4096-bit MODP Group )
+         |
+         +-- Transform PRF ( ID = PRF_HMAC_SHA2_256 )
+         |
+         +-- Transform AKE2 ( ID = SIKE_P434 )
+         |
+         +-- Transform AKE3 ( ID = FRODOKEM_976_AES )
+         |
+         +-- Transform AKE5 ( ID = NONE )
+                ]]></artwork></figure>
+
+                <t> If the initiator includes any Additional Key Exchanges transform types into SA payload in the IKE_SA_INIT exchange request message,
                 it MUST also negotiate using IKE_INTERMEDIATE exchange as described in <xref target="I-D.ietf-ipsecme-ikev2-intermediate" />,
-                by including INTERMEDIATE_EXCHANGE_SUPPORTED notification in the IKE_SA_INIT request message.
-                If the responder agrees to use additional key exchanges, it MUST also return this notification,
+                by including INTERMEDIATE_EXCHANGE_SUPPORTED notification in the same message.
+                If the responder agrees to use additional key exchanges while establishing initial IKE SA, 
+                it MUST also return this notification in the IKE_SA_INIT response message,
                 thus confirming that IKE_INTERMEDIATE exchange is supported and will be used for transferring additional key exchange data.
-                The presence of Additional Key Exchanges transform types in SA payload without negotiation of 
-                using IKE_INTERMEDIATE exchange MUST be treated as protocol error by both initiator and responder.
+                If the IKE_INTERMEDIATE exchange is not negotiated, then the peers MUST treat any Additional Key Exchange transforms
+                in the IKE_SA_INIT exchange messages as unknown transform types and skip the proposals they appear in.
+                If no other proposals are present in the SA payload, the peers will proceed as when no proposal is chosen
```

Comment:
Surely we still have the proposal for Transform Type 4, right? Or did you mean that even both peers cannot agree on the transform ID for Transform Type 4?

Reply:
I meant that AKE Transforms types (not Transform IDs) must be treated as unknown if IKE_INTERMEDIATE wasn't negotiated. So, according to RFC 7296 responder must skip any proposals with unknown transforms. If it finds another proposal with no AKE transform, then it evaluates Transform IDs for every transform in this proposal. If in this proposal Transform Type 4 contains transform ID for PQ KEM, then it's OK, and peers will perform a single PQ Key Exchange in IKE_SA_INIT, with no AKEs.

Reply:
Ok, that makes sense.